src/libs/sfdk/sdk.cpp

```diff
@@ -53,4 +90,59 @@ SdkPrivate::SdkPrivate(Sdk *q)
 
 SdkPrivate::~SdkPrivate() = default;
 
+Utils::FileName SdkPrivate::settingsFile(SettingsScope scope, const QString &basename)
+{
+    const QString prefix = scope == SessionScope
+        ? QString::fromLatin1(Constants::LIB_ID) + '-'
+        : QString();
+    return settingsLocation(scope).appendPath(prefix + basename);
+}
+
+Utils::FileName SdkPrivate::settingsLocation(SettingsScope scope)
+{
+    static FileName systemLocation;
+    static FileName userLocation;
+    static FileName sessionLocation;
+
+    FileName *location = nullptr;
+    switch (scope) {
+    case SystemScope:
+        location = &systemLocation;
+        break;
+    case UserScope:
+        location = &userLocation;
+        break;
+    case SessionScope:
+        location = &sessionLocation;
+        break;
+    }
+    Q_ASSERT(location);
+
+    if (!location->isNull())
+        return *location;
+
+    QTC_CHECK(!QCoreApplication::organizationName().isEmpty());
+    QTC_CHECK(!QCoreApplication::applicationName().isEmpty());
+
+    const QSettings::Scope qscope = scope == SystemScope
+        ? QSettings::SystemScope
+        : QSettings::UserScope;
+    const QString applicationName = scope == SessionScope
+        ? QCoreApplication::applicationName()
+        : QString::fromLatin1(Constants::LIB_ID);
+
+    QSettings settings(QSettings::IniFormat, qscope, QCoreApplication::organizationName(),
+            applicationName);
+
+    // See ICore::userResourcePath()
+    QTC_CHECK(settings.fileName().endsWith(".ini"));
+    const auto iniInfo = QFileInfo(settings.fileName());
+    const QString resourceDir = iniInfo.completeBaseName().toLower();
+    *location = FileName::fromString(iniInfo.path() + '/' + resourceDir);
```

Comment:
I have a feeling this could use some comments. I have no doubt the code above works, and I also believe there are reasons for doing the things that are done, but for the future coder who has to fix bugs here, it might make things faster if here were some explanations. Why are we using QSettings to generate a filename, which we don't even use (but use as a directory name instead). Why are we converting the filename to lower case?

Reply:
There is a comment. Look a bit up.

Reply:
You mean "See ICore::userResourcePath()"? 

I did look at ICore::userResourcePath() before writing my comment. It didn't really answer my questions.

Reply:
But the implementation here is not identical to ICore::userResourcePath(). Looking at ICore::userResourcePath() does not tell me why are we using settings file name as a directory name (userResourcePath() does not do that). It also does not tell me why the filename is turned to lower case before using as a directory name, as userResourcePath does not do that either.

Reply:
identical != similar, (userResourcePath() does not do that) != (code involved in the ICore::userResourcePath() call does not do that)

Reply:
Well, userResourcePath() or any code called by userResourcePath() does not do that. Some code which calls userResourcePath() might do something similar, but maybe you should then hint the reader towards that code? Or maybe just replace the comment with "// see Qt Creator code"? It would be just as helpful...